docs/configure-using-jndi.md

```diff
@@ -63,13 +69,66 @@ The third (and last!) change is to add the reference to
 
 That's it!! (... After some testing, this last step looks unnecessary).
 
-### Deploy the change
+#### Deploy the change
 
 WARNING: `ant clean` is potentially destructive. Know what you are doing.
 
 Perform an `ant clean deploy-war` and restart Tomcat (to pick up the
 JNDI entry). The value defined in JNDI should now be in use by uPortal.
 
+## Define uPortal DataSources
+
+Defining DataSources external to an application was one of the earliest
+uses of JNDI. uPortal has support for JNDI DataSources. In addition,
+a helper factory bean can be configured to use the first working bean.
+ This allows the definition of both a JNDI DataSource bean along with
+ a fallback bean that is defined with filter file values for testing.
+ 
+Example from datasourceContext.xml:
+```aidl
+
+    <!--
+    <bean id="PortalDb" parent="basePooledDataSource">
```

Comment:
Yes, for someone cutting and pasting, the indentation will match the existing file.